users/user-models.js

```diff
@@ -1,35 +1,39 @@
 var mongoose = require('mongoose');
 var Schema = mongoose.Schema;
 
+var options = {discriminatorKey: 'type'};
+
+var userSchema = new Schema({
+      firstName: String,
+      lastName: String,
+      email: String,
+      role: String,
```

Comment:
The information stored in this field will already be stored in the field specified as the discriminator key, so the ```role``` field can be removed from the schema.